App.js

```diff
@@ -0,0 +1,305 @@
+            unit_temp         : '°C',
+            unit_speed        : 'm/s',
+            isFetched         : false
+        };
+    }
+
+    /**
+    * Return user to selected place list
+    * Clear search input
+    */
+    backToList = () => {
+        this.setState({
+            placeSelected : false,
+            placeName     : ''
+        });
+        this.googlePlacesAutocomplete._handleChangeText('');
+    }
+
+    /**
+    * Add selected place to history list
+    */
+    addPlaceToList = () => {
+        this.setState(state => {
+            const dup             = state.selectedPlaceList.find(item => item.name === state.weatherDetails.name);
+            let selectedPlaceList = [];
+
+            if (!dup) 
+                selectedPlaceList = [...state.selectedPlaceList, state.weatherDetails];
+            
+            else
+                selectedPlaceList = [...state.selectedPlaceList];
+            
+
+            return{
+                selectedPlaceList,
+            };
+        });
+    }
+
+    /**
+    * Remove selected place from history list
+    * @param selected place
+    */
+    removePlaceFromList = (place) => {
+        this.setState(state => {
+            const selectedPlaceList = state.selectedPlaceList.filter(item => item.name !== place.name);
+
+            return {
+                selectedPlaceList,
+            };
+        });
+    }
+
+    getAllWeatherDetails = (placeName, unitType) => {
+        let unit_temp      = '';
+        let unit_speed     = '';
+
+        switch(unitType){
+            case 'metric':
+                unit_temp  = '°C';
+                unit_speed = 'm/s';
+            break;
+            case 'default':
+                unit_temp  = 'K';
+                unit_speed = 'm/s';
+            break;
+            case 'imperial':
+                unit_temp  = '°F';
+                unit_speed = 'mph';
+            break;
+        }
+
+        let api_url_weather = 'http://api.openweathermap.org/data/2.5/weather?q='+placeName+'&units='+unitType+'&appid=' + apiKey_openweather;
+
+        fetch(api_url_weather)
+        .then((response) => response.json())
+        .then((responseJson) => {
+            if (responseJson.cod != 404) {
+                iconURI = 'http://openweathermap.org/img/wn/'+responseJson.weather[0].icon+'@2x.png';
+                            
+                this.setState({
+                    listViewDisplayed : false,
+                    placeSelected     : true,
+                    isLoading         : false,
+                    weatherDetails    : {...responseJson, ['unitType']: this.state.unitType},
+                    placeName,
+                    unitType,
+                    unit_temp,
+                    unit_speed,
+
+                }, function(){
+                    this.addPlaceToList(this.state.weatherDetails);
+
+                });
+            }
+            else{
+                this.setState({
```

Comment:
What a reason. you separate `fetch url weather` and `fetch url forecast`.
I think both of two must be success if not alert an error. 
Should it be using only one `Promise chain` to handle error?